pkg/ko/publish/daemon.go

```diff
@@ -47,14 +48,27 @@ func (d *demon) Publish(img v1.Image, s string) (name.Reference, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	tagsAndDigest := append(d.tags, h.Hex)
```

Comment:
This is _probably_ fine in this case, but normally you'd want to use `copy`, then `append` for this.

It's a stupid go thing, but `append` manipulates the underlying array in strange ways that can lead to unexpected behavior: https://allegro.tech/2017/07/golang-slices-gotcha.html

Reply:
Hmm, didn't know about this, as you can see I am actually really new to go. I'll read about.